src/backend/distributed/planner/distributed_planner.c

```diff
@@ -559,7 +559,22 @@ int
 GetRTEIdentity(RangeTblEntry *rte)
 {
 	Assert(rte->rtekind == RTE_RELATION);
-	Assert(rte->values_lists != NIL);
+
+	/*
+	 * RangeTblEntry value_lists must not be empty at this point.
+	 * However, this might happen when processing SQL functions
+	 * in which case we will throw an error.
```

Comment:
Suggestion for a slightly better comment:
```suggestion
	 * Since SQL functions might be in-lined by standard_planner,
	 * we might miss assigning an RTE identity for RangeTblEntry's
	 * related to SQL functions. We already have checks in other
	 * places to throw an error for SQL functions but they are not
	 * sufficient due to function in-lining; so here we capture such
	 * cases and throw an error here.
```